packages/lib-utils/src/config.ts

```diff
@@ -0,0 +1,41 @@
+import type { ResourceFetch } from '@monorepo/common';
+
+export type UtilsConfig = {
+  /**
+   * Resource fetch implementation provided by the host application.
+   *
+   * Applications must validate the response before resolving the Promise.
+   *
+   * If the request cannot be completed successfully, the Promise should be rejected
+   * with an appropriate error.
+   */
+  appFetch: ResourceFetch;
+};
+
+let config: Readonly<UtilsConfig> | undefined;
+
+/**
+ * Set the {@link UtilsConfig} reference.
+ *
+ * This must be done before using any of the Kubernetes utilities.
+ */
+export const setUtilsConfig = (c: UtilsConfig) => {
+  if (config !== undefined) {
+    throw new Error('UtilsConfig reference has already been set');
+  }
+
+  config = Object.freeze({ ...c });
```

Comment:
This ensures that own properties of `UtilsConfig` object aren't modified.

The `getUtilsConfig` function returns the frozen version.